sklearn/cluster/_kmeans.py

```diff
@@ -624,7 +554,6 @@ def _labels_inertia(X, sample_weight, x_squared_norms, centers,
 
     n_threads = _openmp_effective_n_threads(n_threads)
 
-    sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
```

Comment:
This has been done in the main method (fit, predict, etc.)

Reply:
Right. The goal is to do all validations in the fit, predict, etc methods and have all private helpers assume validation is already done, to avoid duplicating validation